python/ycm/vimsupport.py

```diff
@@ -100,7 +100,7 @@ def GetUnsavedAndCurrentBufferData():
       continue
 
     buffers_data[ GetBufferFilepath( buffer_object ) ] = {
-      'contents': '\n'.join( buffer_object ),
+      'contents': '\n'.join( buffer_object ) + '\n',
```

Comment:
Hm, this will create another copy of the entire buffer just so that it can append one char.

Anyway, add a comment above this line mentioning why this extra newline is needed. Something like:

`# Newline needed to prevent spurious warning. See #1455 for details.`